### PR TITLE
fix: Inline editing flaky test fix

### PR DESCRIPTION
**Pull request by sagar-qa007**
## Description

EE PR: [Appsmith EE Pull Request #5158](https://github.com/appsmithorg/appsmith-ee/pull/5158)

Root Cause Analysis (RCA):
The `updateCodeInput` function was identified as a source of flakiness due to improper execution. Additionally, the `SaveLocalStorageCache` caused issues by retaining the reset button after the 6th test case, resulting in a duplicate button in the 7th test case, which led to test failures.

Solution:
Replaced `updateCodeInput` with `EnterJSContext` from `agHelper` for more reliable execution. For the second flaky step, a new page was created, and a table with relevant data was added. This resolved the issue without requiring significant changes to the test case or workflow.

Fixes #`36365`  

## Automation

/ok-to-test tags="@tag.ImportExport"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10940196158>
> Commit: 32d1be4c3dfe54bf8a0f8e9f54b20d2845c86407
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10940196158&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.ImportExport`
> Spec:
> <hr>Thu, 19 Sep 2024 12:01:32 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Bug Fixes**
	- Adjusted the inline editing test suite for the table widget to improve state management between tests.

- **New Features**
	- Updated the limited tests to focus on inline editing functionality, enhancing testing priorities for the Table V2 widget.
	- Introduced new properties in the widget locators to enhance UI interactions, including selectors for various controls and elements within the property pane and cell editing functionalities.
	- Added a new JavaScript toggle selector to facilitate interactions in tests.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js [hunk 1]
```diff
@@ -161,6 +157,7 @@ describe(
     });
 
     it("7. should check if updatedRowIndex is getting updated for multi row update mode", () => {
+      agHelper.AddDsl("Table/InlineEditingDSL");
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js:160]: this is already in beforeEach. why do this again inside the test? @sagar-qa007

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js (hunk 1 above)
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js]: No need for it but by adding this line fixed the issues. It is not needed but then found another flaky step so fix it as well. Let's see new results.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js [hunk 2]
```diff
@@ -171,9 +175,11 @@ describe(
       cy.get(".t--widget-buttonwidget").should("exist");
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js:175]: can you add these to locators and use it please? all of it in the test. Let's fix the test we touch completely @sagar-qa007

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js (hunk 2 above)
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js:175]: Sure. Let me update please.

File: app/client/cypress/limited-tests.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
+cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_2_spec.js
```
Comment (ApekshaBhosale) [inline app/client/cypress/limited-tests.txt:2]: revert this change

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

**Comment by sagar-qa007** [comment]
@ApekshaBhosale Changes are reverted for limited test. Please review now.

**Review by yatinappsmith — APPROVED** [review]

**Pull request merged**